src/formatDate.js

```diff
@@ -50,7 +91,49 @@
  */
 
 function formatDate(date, fromFormat, toFormat) {
-  // write code here
+  const dateParts = date.split(fromFormat[fromFormat.length - 1]);
+
+  const formatMap = {};
+
+  fromFormat.forEach((token, index) => {
+    formatMap[token] = dateParts[index];
+  });
+
+  const convertedParts = [];
+
+  toFormat.forEach(token => {
```

Comment:
Instead of using a traditional function in the forEach, consider using an arrow function for more concise, modern JavaScript.